provider/source/src/cldr_serde/units/info.rs

```diff
@@ -92,24 +92,16 @@ impl Resource {
     /// * `Ok(u16)` - The unique identifier for the unit if found.
     /// * `Err(DataError)` - An error if the unit is not found or if the index is out of range for `u16`.
     pub fn unit_id(&self, unit_name: &str) -> Result<u16, DataError> {
-        let mut cursor = unit_name.as_bytes().iter().try_fold(
-            DECIMAL_PREFIXES_TRIE.cursor(),
-            |mut cursor, &byte| {
-                cursor.step(byte);
-                if cursor.is_empty() {
-                    Err(DataError::custom("Unit not found"))
-                } else {
-                    Ok(cursor)
-                }
-            },
-        )?;
-
-        cursor.take_value().map_or_else(
-            || Err(DataError::custom("Unit not found")),
-            |value| {
+        let mut cursor = CLDR_IDS_TRIE.cursor();
+        cursor
+            .write_str(unit_name)
+            .map_err(|_| DataError::custom("Unit not found"))?;
```

Comment:
nit: this is infallible if the id is ascii, maybe a better error message?

Reply:
now, the error message is correct.